src/hypothesis/strategies.py

```diff
@@ -1575,4 +1578,96 @@ def deferred(definition):
     return DeferredStrategy(definition)
 
 
+class _PathLike(object):
+    """A class implementing the os.PathLike protocol available since Python
+    3.6.
+
+    See https://www.python.org/dev/peps/pep-0519/ for more details.
+    """
+
+    def __init__(self, value):
+        self._value = value
+
+    def __fspath__(self):
+        return self._value
+
+    def __repr__(self):
+        return 'pathlike(%r)' % self._value
+
+
+@defines_strategy
+@composite
+def fspaths(draw, allow_pathlike=True, allow_existing=False):
+    """A strategy which generates filesystem path values.
+
+    The generated values include everything which the builtin
+    :func:`python:open` function accepts i.e. which won't lead to
+    :exc:`ValueError` or :exc:`TypeError` being raised.
+
+    Note that the range of the returned values depends on the operating
+    system, the Python version, and on the filesystem encoding as returned by
+    :func:`sys.getfilesystemencoding`.
+
+    :param bool allow_pathlike:
+        If the result can be a pathlike (see :class:`os.PathLike`)
+    :param bool allow_existing:
+        If paths which happen to exist on the filesystem should be returned.
+        This is :obj:`python:False` by default to prevent tests from accessing
+        existing files by accident.
+
+    .. versionadded:: 3.15
+
+    """
+
+    strategies = []
+
+    if os.name == 'nt':  # pragma: no branch
```

Comment:
Yeah https://coverage.readthedocs.io/en/coverage-4.3.4/excluding.html#branch-coverage

so "no branch" will exclude just the marked case and imply "no branch" if I understand correctly.